pluginframework/data_volumes.go

```diff
@@ -0,0 +1,95 @@
+package pluginframework
```

Comment:
Suggestion for code organization: let's make a new package for each resource/data source. Datasource(s) and resource for the same entity can share the package. e.g. internal/resources/volume/... defines DataSourceVolume() and ResourceVolume(),  internal/resources/qualitymonitor/... defines ResourceQualityMonitor(), etc.